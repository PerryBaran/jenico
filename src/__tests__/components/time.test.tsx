import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Time from "../../components/player/time/Time";

describe("Time", () => {
  const validProps = {
    time: 20
  };

  test("snapshot", () => {
    const { asFragment } = render(<Time time={validProps.time} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe("correctly renders time", () => {
    test("time === 0", () => {
      render(<Time time={0} />);

      expect(screen.getByText("0:00")).toBeInTheDocument();
    });

    test("time === 30", () => {
      render(<Time time={30} />);

      expect(screen.getByText("0:30")).toBeInTheDocument();
    });

    test("time === 60", () => {
      render(<Time time={60} />);

      expect(screen.getByText("1:00")).toBeInTheDocument();
    });

    test("time === 600", () => {
      render(<Time time={600} />);

      expect(screen.getByText("10:00")).toBeInTheDocument();
    });
  });
});