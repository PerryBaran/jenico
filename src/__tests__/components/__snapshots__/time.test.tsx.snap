// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p>
      0:20
    </p>
  </div>
</DocumentFragment>
`;
